File: TranslateSS.py
def dec_to_base(N, base):
    if not hasattr(dec_to_base, 'table'):
        dec_to_base.table = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    x,y = divmod(N, base)
    return dec_to_base(x, base) + dec_to_base.table[y] if x else dec_to_base.table[y]


key3 = None
while key3 != True:
    try:
        first = int(input("Из какой системы счисления надо будет переводить? - "))
        if first >= 2 and first <= 36:
            key3 = True
        else:
            print("Введи целое число от 2 до 36")
    except:
        print("Введи целое число от 2 до 36")

key2 = None
while key2 != True:
    try:
        second = int(input("В какую систему счисления надо будет переводить? - "))
        if second >= 2 and second <= 36:
            key2 = True
        else:
            print("Введи целое число от 2 до 36")
    except:
        print("Введи целое число от 2 до 36")

key = None
while key != True:
    x = input("Какое число надо будет переводить? - ")
    try:
        g = int(x, first)
        key = True
    except ValueError:
        print("Введи любое целое число. Примечание: Цифры в числе не должны быть меньше системы счисления, из которой переводишь")


print(dec_to_base(g, second))


input()

File: TranslateSS.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

// Функция перевода десятичного числа в указанную систему счисления
func decToBase(n int, base int) string {
	const table = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
	if n < base {
		return string(table[n])
	}
	return decToBase(n/base, base) + string(table[n%base])
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	var fromBase, toBase int

	// Ввод начальной системы счисления
	for {
		fmt.Print("Из какой системы счисления надо будет переводить? - ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		n, err := strconv.Atoi(input)
		if err == nil && n >= 2 && n <= 36 {
			fromBase = n
			break
		}
		fmt.Println("Введи целое число от 2 до 36")
	}

	// Ввод конечной системы счисления
	for {
		fmt.Print("В какую систему счисления надо будет переводить? - ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		n, err := strconv.Atoi(input)
		if err == nil && n >= 2 && n <= 36 {
			toBase = n
			break
		}
		fmt.Println("Введи целое число от 2 до 36")
	}

	// Ввод числа и перевод в десятичную систему
	var value int
	for {
		fmt.Print("Какое число надо будет переводить? - ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		n, err := strconv.ParseInt(input, fromBase, 64)
		if err == nil {
			value = int(n)
			break
		}
		fmt.Println("Введи любое целое число. Примечание: цифры в числе не должны быть меньше системы счисления, из которой переводишь")
	}

	// Вывод результата
	fmt.Println("Результат:", decToBase(value, toBase))

	// Пауза
	fmt.Println("\nНажмите ENTER для выхода.")
	reader.ReadString('\n')
}
